### PR TITLE
M, E edge case fix

### PR DESCRIPTION
**Pull request by icweaver**
PR mentioned in https://github.com/JuliaAstro/Transits.jl/pull/6#issuecomment-791370024 to address an edge case test

### CODE REVIEW

File: src/kepler_solver.jl [hunk 1]
```diff
@@ -4,10 +4,12 @@
 function kepler_solver(M::T, e::T) where {T<:AbstractFloat}
     @assert 0 <= e <= 1 "eccentricity must be in the range [0, 1]"
     # M must be in the range [-pi, pi], see Markley (1995), page 2.
-    M = rem2pi(M, RoundNearest)
+    M = M % (2.0*π) == 0.0 ? 0.0 : mod2pi(M)
```
Comment (giordano) [inline src/kepler_solver.jl:7]: I don't understand this:
```julia
julia> rem2pi(2pi, RoundNearest)
-2.4492935982947064e-16
```
It isn't _exactly_ zero, but it isn't wrong either, is it?  I odn't like too much manually flushing to 0

Comment (icweaver) [reply src/kepler_solver.jl:7]: Hey Mosè, thanks for getting back to me so quickly!

Right, so I think the problem is more in terms of the sign of M. `rem2pi` returns a negative value in the example you gave while `mod2pi` does not, like you mentioned. This seems to cause side effects in how some of the other intermediate values are calculated, ultimately leading to drastically different results when trying something like (M, e) = (-226.19467108901554, 0.999999), which returns -0.0053289420875433135 while kepler.py returns and 6.2778563645118775.

> which forces you to do the high dance below
hahaha, I am more than happy to do any additional dancing to not be as stringent with the flushing to zero, this is just a working example I came up with

Comment (giordano) [reply src/kepler_solver.jl:7]: So perhaps you want to compare
```julia
julia> mod2pi(-0.0053289420875433135) ≈ 6.2778563645118775
true
```
?

Comment (icweaver) [reply src/kepler_solver.jl:7]: Since we're just off by 2pi, maybe that can just be fixed further downstream?

Comment (giordano) [reply src/kepler_solver.jl:7]: But I'm a bit confused: [`kepler_solver`](http://juliaastro.github.io/AstroLib.jl/stable/ref.html#AstroLib.kepler_solver-Tuple{Real,Real}) is documented to return values in `[−π, π]`, how do you expect to get `6.2778563645118775`?

Comment (giordano) [reply src/kepler_solver.jl:7]: It isn't clear to me what you're _exactly_ comparing.  In terms of trigonometric functions, it shouldn't matter much.  If you're comparing side-by-side the values returned by two different Kepler solvers which return values in different ranges, you need to reduce one to the other range in any case

Comment (icweaver) [reply src/kepler_solver.jl:7]: Right, I am doing a correct comparison. I stepped line-by-line through both solvers and they look to be doing the same thing. I still need to go through the theory paper, but is there a physical interpretation to why we would expect the bound s to be [-pi, pi] on our output?

Comment (giordano) [reply src/kepler_solver.jl:7]: Quoting from the paper, page 2 (referenced also in the comments):

> It is assumed that E and M have been reduced by multiples of 2π to have absolute
value less than or equal to π.

I'm pretty sure in my initial attempts I had them in the wrong range and got wrong results

Comment (giordano) [reply src/kepler_solver.jl:7]: Also, if we really want to check who gives the most accurate solution for the equation:
```julia
julia> E_astrolib = -0.0053289420875433135
-0.0053289420875433135

julia> E_keplerpy = 6.2778563645118775
6.2778563645118775

julia> M = -226.19467108901554
-226.19467108901554

julia> rem2pi(M, RoundNearest) - (E_astrolib - e * sin(E_astrolib))
-1.4247358870613349e-19

julia> mod2pi(M) - (E_keplerpy - e * sin(E_keplerpy))
8.881784197001252e-15
```
:slightly_smiling_face: (but frankly, the difference at this level is negligible)

Comment (icweaver) [reply src/kepler_solver.jl:7]: Thanks for the checks, this really helped a lot! I also appreciate the reference to the bounds used in the text you sent, it's just kind of a bummer that the reasoning the author gives is sort of circular (no pun intended). I'm sure there is a good reason somewhere why the `rem2pi` docs state that [-pi, pi] tends to be more numerically accurate.

At any rate, sanitizing the test inputs first with it fixes all the tests now, so I'm happy to close this issue. Thanks for taking the time to humor me!

Comment (giordano) [reply src/kepler_solver.jl:7]: It was a good discussion :slightly_smiling_face:  I'm going to close the pull request
